### PR TITLE
Fixed response code for `mergeCertificate` operation.

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes: #36260. Also added a test, as we were lacking one for this particular operation.

NOTE: While adding tests I noticed I had to modify some configuration using the `maven-resources-plugin` to have it properly load our test `.der` certificate.

### CODE REVIEW

**Review by heaths — COMMENTED** [review]
The impl looks correct, according to https://github.com/Azure/azure-rest-api-specs/blob/e45656f49582887382dec9d3c31bc661796a4fce/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.5-preview.1/certificates.json#L968. I can't really comment on the rest, though.

File: sdk/keyvault/azure-security-keyvault-certificates/pom.xml [hunk 1]
```diff
@@ -119,5 +110,38 @@
       <version>4.11.0</version> <!-- {x-version-update;org.mockito:mockito-core;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-certificates/pom.xml:114]: While this is a test dependency, not a fan of bouncy castle, due to its CVE issues and can influence users to use Bouncy castle as the code is tested with it.

As a follow up investigation, would be good to replace with an alternative library.

**Review by g2vinay — APPROVED** [review]
Approving as the bug fix is correct.
Would be good to replace Bouncy castle with an alternative (to be investigated).

**Pull request merged**